vector/src/main/java/im/vector/app/features/home/NewHomeDetailFragment.kt

```diff
@@ -201,13 +201,12 @@ class NewHomeDetailFragment :
     private fun setupFabs() {
         showFABs()
 
-        views.newLayoutCreateChatButton.setOnClickListener {
-            newChatBottomSheet.show(requireActivity().supportFragmentManager, NewChatBottomSheet.TAG)
+        views.newLayoutCreateChatButton.debouncedClicks {
+            newChatBottomSheet.takeIf { !it.isAdded }?.show(requireActivity().supportFragmentManager, NewChatBottomSheet.TAG)
         }
 
-        views.newLayoutOpenSpacesButton.setOnClickListener {
-            // Click action for open spaces modal goes here
-            spaceListBottomSheet.show(requireActivity().supportFragmentManager, SpaceListBottomSheet.TAG)
+        views.newLayoutOpenSpacesButton.debouncedClicks {
+            spaceListBottomSheet.takeIf { !it.isAdded }?.show(requireActivity().supportFragmentManager, SpaceListBottomSheet.TAG)
```

Comment:
I think the root cause of the issue is to instantiate the BottomSheets in advance [at this point](https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/home/NewHomeDetailFragment.kt#L93). I would have fixed this issue differently. Using `debouncedClicks` is OK, but using `takeIf { !it.isAdded }` looks a bit like a workaround.

WDYT?

Reply:
Also there is no need to inject the constructor [here](https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/home/room/list/home/NewChatBottomSheet.kt#L30).

Reply:
I believe the approach you're leading towards is to instantiate the bottom sheet as we show it? (`SpaceListBottomSheet.show()`)

I considered this approach but doing so would just reduce the bug from a crash to having two bottom sheets show together. `debouncedClicks` may help with this, but I wanted to add the `!it.isAdded` check as a guaranteed method.

Research also shows me that making this check (or using `bottomSheet.isShowing` in other cases) is a common practice.

Reply:
One more point:
When `spaceListBottomSheet` is displayed, if the user rotates it's device, the Fragment `SpaceListBottomSheet` will be restored by the System (the BottomSheet will still be visible), but `NewHomeDetailFragment` will not have a correct reference to this Fragment, since it will just instantiate a new instance in the constructor.

So this code:
https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/home/NewHomeDetailFragment.kt#L197

will not work (I did not check though).

You should try to get the reference (by tag for instance) to the restored bottom sheet Fragment instead of creating a new one.